File: app/src/main/java/fi/kaupunkifillarit/model/Rack.java
package fi.kaupunkifillarit.model;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

@JsonObject
public class Rack {
    @JsonField
    public String id;
    @JsonField
    public String name;
    @JsonField(name = "lat")
    public double latitude;
    @JsonField(name = "lon")
    public double longitude;
    @JsonField(name = "bikesAvailable")
    public int bikes;
    @JsonField(name = "spacesAvailable")
    public int spaces;

    public Rack() { }

    public Rack(String id,
                String name,
                double latitude,
                double longitude,
                int bikes,
                int spaces) {
        this.id = id;
        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
        this.bikes = bikes;
        this.spaces = spaces;
    }

    public int getSlots() {
        return bikes + spaces;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Rack tram = (Rack) o;

        return id.equals(tram.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/model/Rack.kt
package fi.kaupunkifillarit.model

import com.bluelinelabs.logansquare.annotation.JsonField
import com.bluelinelabs.logansquare.annotation.JsonObject

@JsonObject
class Rack {
    @JsonField
    var id: String = ""
    @JsonField
    var name: String = ""
    @JsonField(name = arrayOf("lat"))
    var latitude: Double = 0.0
    @JsonField(name = arrayOf("lon"))
    var longitude: Double = 0.0
    @JsonField(name = arrayOf("bikesAvailable"))
    var bikes: Int = 0

    constructor() {
    }

    constructor(id: String,
                name: String,
                latitude: Double,
                longitude: Double,
                bikes: Int) {
        this.id = id
        this.name = name
        this.latitude = latitude
        this.longitude = longitude
        this.bikes = bikes
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val tram = other as Rack?

        return id == tram!!.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}
